src/main/java/net/snowflake/client/jdbc/SnowflakeFileTransferAgent.java

```diff
@@ -1200,6 +1209,18 @@ static StageInfo getStageInfo(JsonNode jsonNode, SFSession session) throws Snowf
     return stageInfo;
   }
 
+  private static void setupUseRegionalUrl(JsonNode jsonNode, StageInfo stageInfo) {
+    if (stageInfo.getStageType() != StageInfo.StageType.GCS
+        && stageInfo.getStageType() != StageInfo.StageType.S3) {
```

Comment:
So this is also for S3? I had an impression it is for GCS only.

Reply:
we plan to migrate to `useRegionalUrl` instead of explicit `useS3RegionalUrl` in SNOW-1818803